nncf/experimental/torch/sparsity/movement/algo.py

```diff
@@ -0,0 +1,201 @@
+from nncf.torch.algo_selector import PT_COMPRESSION_ALGORITHMS
+from nncf.torch.compression_method_api import PTCompressionAlgorithmController
+from nncf.torch.graph.transformations.commands import PTInsertionCommand
+from nncf.torch.graph.transformations.commands import PTTargetPoint
+from nncf.torch.graph.transformations.commands import TransformationPriority
+from nncf.torch.layers import NNCFLinear
+from nncf.torch.module_operations import UpdateWeightAndBias
+from nncf.torch.nncf_network import NNCFNetwork
+from nncf.torch.sparsity.base_algo import BaseSparsityAlgoBuilder
+from nncf.torch.sparsity.base_algo import BaseSparsityAlgoController
+from nncf.torch.sparsity.base_algo import SparseModuleInfo
+from nncf.torch.sparsity.collector import PTSparseModelStatisticsCollector
+from nncf.torch.utils import get_model_device
+
+SUPPORTED_NNCF_MODULES = [NNCFLinear]
+
+
+@PT_COMPRESSION_ALGORITHMS.register('movement_sparsity')
+class MovementSparsityBuilder(BaseSparsityAlgoBuilder):
+    def __init__(self, config, should_init: bool = True):
+        super().__init__(config, should_init)
+        configs = self._algo_config.get('sparse_structure_by_scopes', [])
+        self._sparse_configs_by_scopes = [SparseConfigByScope.from_config(c) for c in configs]
+
+    def create_weight_sparsifying_operation(self, target_module_node: NNCFNode,
+                                            compression_lr_multiplier: float) -> MovementSparsifier:
+        sparse_cfg = SparseConfig(SparseStructure.FINE)
+        node_name = target_module_node.node_name
+        matched_scopes = []
+        for configs_per_scopes in self._sparse_configs_by_scopes:
+            target_scopes = configs_per_scopes.target_scopes
+            if matches_any(node_name, target_scopes):
+                sparse_cfg = configs_per_scopes.sparse_config
+                matched_scopes.append(target_scopes)
+        if len(matched_scopes) >= 2:
+            raise RuntimeError(f'"{node_name}" is matched by multiple items in `sparse_structure_by_scopes`.')
+
+        return MovementSparsifier(target_module_node, sparse_cfg=sparse_cfg, frozen=False,
+                                  compression_lr_multiplier=compression_lr_multiplier,
+                                  layerwise_loss_lambda=0.5)
+
+    def _sparsify_weights(self, target_model: NNCFNetwork) -> List[PTInsertionCommand]:
+        device = get_model_device(target_model)
+        sparsified_module_nodes = target_model.get_weighted_original_graph_nodes(
+            nncf_module_names=[m.__name__ for m in SUPPORTED_NNCF_MODULES])
+        insertion_commands = []
+        for module_node in sparsified_module_nodes:
+            node_name = module_node.node_name
+
+            if not self._should_consider_scope(node_name):
+                logger.info("Ignored adding Weight Sparsifier in scope: {}".format(node_name))
+                continue
+
+            logger.info("Adding Weight Sparsifier in scope: {}".format(node_name))
```

Comment:
Please be aware that we are cleaning messages that are printed by NNCF.
https://github.com/openvinotoolkit/nncf/pull/1418/files#diff-cdaea7f5c2f1342e1fc839334b0391ae5fad9e8e3075fa48719590e7227e41a0R762

Please follow introduced guidelines in the PR.
For example, this particular message should be moved to debug level.